deployability/modules/allocation/aws/provider.py

```diff
@@ -81,13 +81,16 @@ def _create_instance(cls, base_dir: Path, params: CreationPayload, config: AWSCo
                 name = repository + "-" + str(re.search(r'(\d+)$', issue).group(1)) + "-" + str(params.composite_name.split("-")[1]) + "-" + str(params.composite_name.split("-")[2])
 
             # Keys.
-            if not ssh_key:
-                logger.debug(f"Generating new key pair")
-                credentials.generate(temp_dir, str('-'.join(name.split("-")[:-2])))
+            if platform == "windows":
+                credentials.create_password()
             else:
-                logger.debug(f"Using provided key pair")
-                key_id = credentials.ssh_key_interpreter(ssh_key)
-                credentials.load(key_id)
+                if not ssh_key:
+                    logger.debug(f"Generating new key pair")
+                    credentials.generate(temp_dir, str('-'.join(name.split("-")[:-2])))
+                else:
+                    logger.debug(f"Using provided key pair")
+                    key_id = credentials.ssh_key_interpreter(ssh_key)
+                    credentials.load(key_id)
```

Comment:
This could be a bit simplified by just using an elif. Something like
```suggestion
            elif not ssh_key:
                logger.debug(f"Generating new key pair")
                credentials.generate(temp_dir, str('-'.join(name.split("-")[:-2])))
            else:
                logger.debug(f"Using provided key pair")
                key_id = credentials.ssh_key_interpreter(ssh_key)
                credentials.load(key_id)
```

Reply:
Done: f1fb1e8fa